extensions/resource-deployment/src/services/tools/azCliTool.ts

```diff
@@ -40,8 +43,86 @@ export class AzCliTool extends ToolBase {
 		return true;
 	}
 
-	public install(): Promise<void> {
-		return Promise.reject('not implemented');
+	protected get installationPath(): Promise<string | null> {
+		return new Promise<string | null>((resolve, _reject) => {
+			switch (this.osType) {
+				case OsType.linux:
+					resolve(defaultInstallationRoot);
+				default:
+					resolve(win32InstallationRoot);
+					console.log(`TCL: AzCliTool -> win32InstallationRoot:${win32InstallationRoot}`);
+			}
+		});
+	}
+	get installationCommands(): Command[] {
```

Comment:
Nit. Instead of these being in the code can they be in a osName.installation.json file? That way we're keeping stuff like this that can change out of the code and we don't have to have massive switch - cases

Reply:
I have always had debates with folks whether these kinds of things should be in configuration files or in code. The argument against keeping in configuration file was always that it moves the code away from the execution place which makes it harder to see what is happenning and thus making th code harder to maintian because when changes need to be made one has to look up more to figure out where the change needs to go instead of the change just going in the correspond ITool.ts file.

 If your concern is with massive switch statement then I can change it a constant object that is indexed by osType to give the apropriate installationCommands.

In any case we all prefer to keep these in out of band json files then I can make the change, so everyone kindly indicate your preferences.